src/components/common/FieldForm.js

```diff
@@ -0,0 +1,187 @@
+    if (reason === 'clickaway') {
+      return;
+    }
+    setAlertStatus(false);
+  };
+
+  const showAlert = (message, severity) => {
+    setAlertMessage(message);
+    setAlertSeverity(severity);
+    setAlertStatus(true);
+  };
+
+  const handleTextChange = ({ target }) => {
+    const { name, value } = target;
+    if (name === 'fieldId') {
+      setFieldIdentifier(value);
+    } else if (name === 'area') {
+      setArea(value);
+    }
+  };
+
+  const handleSubmit = () => {
+    let payload = {};
+    // payload.name = fieldIdentifier;
+    payload.identifier = fieldIdentifier;
+    payload.location = locality;
+    payload.area = Number(area);
+    payload.id = selectedRow.obj.id;
+    
+    // console.log(selectedRow.obj);
+    if (operation === 'UPDATE') {
+      updateField(payload)
+        .then((response) => {
+          // showAlert('Field successfully updated', 'info');
+          closeHandler();          
+        })
+        .catch((e) => {
+          console.log('Internal server error', e);
+          showAlert('Field updation failed: ' + e.message, 'error');
+        });
+    } else {
+      createField(payload)
+        .then((response) => {
+          showAlert('Field successfully created', 'info');
+        })
+        .catch((e) => {
+          console.log('Internal server error', e);
+          showAlert('Field creation failed: ' + e.message, 'error');
+        });
+    }
+  };
+
+  const handleChange = ({ target }) => {
+    const { name, value } = target;
+    if (name === 'locality') {
+      setLocality(value);
+    }
+  };
+
+  return (
+    <FormGroup className={classes.formGroup}>
+      <Typography align='center' variant='h6' className={classes.title}>
+        {title}
+      </Typography>
+
+      <FormControl className={classes.formControl}>
+        <InputLabel id='locality-label'>Locality</InputLabel>
+        <Select
+          id='locality'
+          name='locality'
+          value={locality}
+          onChange={handleChange}
+        >
+          {locations.map((location) => {
+            return (
+              <MenuItem
+                key={location.code}
+                value={location.code}
+                className={classes.menuItem}
+              >
+                {location.displayStr}
+              </MenuItem>
+            );
+          })}
+        </Select>
+      </FormControl>
+
+      <TextField
+        className={classes.formControl}
+        id='fieldId'
+        name='fieldId'
+        onChange={handleTextChange}
+        label='Field Name'
+        value={fieldIdentifier}
+      />
+
+      <TextField
+        className={classes.formControl}
+        id='area'
+        name='area'
+        onChange={handleTextChange}
+        label='Area'
+        value={area}
+      />
+
+      <Button
+        variant='contained'
+        color='primary'
+        size='medium'
+        className={classes.submitButton}
+        onClick={handleSubmit}
+      >
+        Submit
```

Comment:
Let's take this also as a prop in the component. We can use `Save` for update and `Create` for creation.